code/game/gamemodes/objective.dm

```diff
@@ -116,37 +116,40 @@ GLOBAL_LIST_EMPTY(objectives)
 	var/list/datum/mind/owners = get_owners()
 	if(!dupe_search_range)
 		dupe_search_range = get_owners()
-	var/list/possible_targets = list()
-	var/try_target_late_joiners = FALSE
-	for(var/I in owners)
-		var/datum/mind/O = I
-		if(O.late_joiner)
-			try_target_late_joiners = TRUE
-	for(var/datum/mind/possible_target in get_crewmember_minds())
-		var/target_area = get_area(possible_target.current)
-		if(possible_target in owners)
-			continue
-		if(!ishuman(possible_target.current))
-			continue
-		if(possible_target.current.stat == DEAD)
-			continue
-		if(!is_unique_objective(possible_target,dupe_search_range))
-			continue
-		if(!HAS_TRAIT(SSstation, STATION_TRAIT_LATE_ARRIVALS) && istype(target_area, /area/shuttle/arrival))
-			continue
-		if(possible_target in blacklist)
-			continue
-		possible_targets += possible_target
-	if(try_target_late_joiners)
-		var/list/all_possible_targets = possible_targets.Copy()
-		for(var/I in all_possible_targets)
-			var/datum/mind/PT = I
-			if(!PT.late_joiner)
-				possible_targets -= PT
-		if(!possible_targets.len)
-			possible_targets = all_possible_targets
-	if(possible_targets.len > 0)
-		target = pick(possible_targets)
+	if(SSstation.protagonists.len && prob(80))
+		target = pick(SSstation.protagonists)
+	else
+		var/list/possible_targets = list()
+		var/try_target_late_joiners = FALSE
+		for(var/I in owners)
```

Comment:
extra credit, if you want: this is all old objective code, so if you could apply the "as anything" to for loops iterating through types it knows and also apply the "descriptive variables" standard of naming variables, that'd be sick